packages/x-tree-view/src/internals/utils/plugins.ts

```diff
@@ -0,0 +1,12 @@
+import { TreeViewAnyPluginSignature, TreeViewInstance, TreeViewPlugin } from '../models';
+
+export const hasPlugin = <
```

Comment:
I can't have this inside `useTreeViewOptionalPlugins` because `instance` needs to be a param in order to infer its shape.

Otherwise we would need to do `instance.hasPlugin(instance, plugin)` which is worse IMHO.